File: step-definitions/shared/data-loader.js
const path = require('path');

function loadFixture(fileName, key) {
  const filePath = path.join(process.cwd(), 'fixtures', `${fileName}.json`);
  delete require.cache[require.resolve(filePath)];
  const data = require(filePath);
  if (key) {
    if (!Object.prototype.hasOwnProperty.call(data, key)) {
      throw new Error(`Fixture key "${key}" not found in fixtures/${fileName}.json`);
    }
    return data[key];
  }
  return data;
}

module.exports = { loadFixture };

File: utils/data_loader.py
import json
import os


def load_fixture(file_name, key=None):
    file_path = os.path.join(os.getcwd(), 'fixtures', f'{file_name}.json')
    with open(file_path) as f:
        data = json.load(f)
    if key is not None:
        if key not in data:
            raise KeyError(f'Fixture key "{key}" not found in fixtures/{file_name}.json')
        return data[key]
    return data
